### PR TITLE
Add API Key authentication docs

### PR DESCRIPTION
**Pull request by phillipleblanc**
## 🗣 Description

Add documentation on how to use API key authentication with Spice endpoints.

### CODE REVIEW

File: spiceaidocs/docs/api/auth/index.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      keys:
+        - ${ secrets:api_key } # Use the secret replacement syntax to load the API key from a secret store
+        - 1234567890 # Or specify the API key directly
+```
+
+To learn more about secrets, see [Secret Stores](../../components/secret-stores/index.md).
+
+:::info
+
+The API key authentication is applied on startup and changes will not take effect until the runtime is restarted.
```
Comment (eadgbear) [inline spiceaidocs/docs/api/auth/index.md:28]:
```suggestion
The API key authentication is applied only on startup and changes will not take effect until the runtime is restarted.
```

**Review by eadgbear — APPROVED** [review]

File: spiceaidocs/docs/api/auth/index.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+Spice supports adding optional authentication to its API endpoints via configurable API keys.
+
+Use the `auth` section as a child to `runtime` to provide the API keys. Multiple API keys can be specified, and any of the keys can be used to authenticate requests.
+
+```yaml
+runtime:
+  auth:
+    api-key:
+      enabled: true
+      keys:
```
Comment (Jeadie) [inline spiceaidocs/docs/api/auth/index.md:19]: Does this work?
```yaml
keys: ${ secrets:api_key_list }
```
`export api_key_list="['foo','bar']"`

File: spiceaidocs/docs/api/auth/index.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+The API key authentication is applied on startup and changes will not take effect until the runtime is restarted.
+
+:::
+
+## HTTP
+
+For HTTP routes, the API key is expected to be included in the `X-API-Key` header.
+
+```bash
```
Comment (Jeadie) [inline spiceaidocs/docs/api/auth/index.md:36]: I though HTTP endpoints support `Authorization: Bearer ${ api_key }` too?

File: spiceaidocs/docs/api/auth/index.md (hunk 2 above)
Comment (phillipleblanc) [reply spiceaidocs/docs/api/auth/index.md:19]: No

File: spiceaidocs/docs/api/auth/index.md (hunk 3 above)
Comment (phillipleblanc) [reply spiceaidocs/docs/api/auth/index.md:36]: No, they only look at the `X-API-Key` header. The Flight endpoint looks at the `Authorization` header to comply with the Flight BasicAuth pattern, but not the HTTP endpoints.

**Pull request merged**